presto-main/src/main/java/io/prestosql/sql/planner/planprinter/IoPlanPrinter.java

```diff
@@ -650,6 +663,22 @@ else if (writerTarget instanceof CreateReference || writerTarget instanceof Inse
             return processChildren(node, context);
         }
 
+        private void addInputTableConstraints(TupleDomain<ColumnHandle> filterDomain, TableScanNode tableScanNode, IoPlanBuilder context)
+        {
+            TableHandle table = tableScanNode.getTable();
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, table);
+            TupleDomain<ColumnHandle> predicate = metadata.getTableProperties(session, table).getPredicate();
+            EstimatedStatsAndCost estimatedStatsAndCost = getEstimatedStatsAndCost(tableScanNode);
+            context.addInputTableColumnInfo(
+                    new IoPlan.TableColumnInfo(
+                            new CatalogSchemaTableName(
+                                    tableMetadata.getCatalogName().getCatalogName(),
+                                    tableMetadata.getTable().getSchemaName(),
+                                    tableMetadata.getTable().getTableName()),
+                            Sets.union(parseConstraints(table, predicate), parseConstraints(table, filterDomain)),
```

Comment:
What if there's an overlap between the table properties and the filter? (e.g., a filter limits a column to some range and table properties limit the same column to another range)? If I read this correctly, you'll end up with two entries for that column. This code should do an intersection of the corresponding domains.